[SQLLab, CSV export] UnicodeEncodeError when tab/filename contains non-ascii characters

Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if any
- [x] I have reproduced the issue with at least the latest released version of superset
- [x] I have checked the issue tracker for the same issue and I haven't found one similar


### Superset version
0.20.6

### Expected results
Exporting a query in SQLLab to CSV works regardless of the presence of non-ascii characters in the tab label.

### Actual results
If a tab cointains non-ascii characters , like e.g. 'ü', the following error is thrown:

```
2017-11-29 09:40:07,000:INFO:root:Exporting CSV file [Hkqwzx3gM]
2017-11-29 09:40:07,004:INFO:root:Parsing with sqlparse statement SELECT 1
2017-11-29 09:40:07,004:INFO:root:Fetching CSV from results backend [41ae1238-57cf-4a5a-ba21-7ff2776062d3]
2017-11-29 09:40:07,005:INFO:root:Decompressing
2017-11-29 09:40:07,006:INFO:root:Using pandas to convert to CSV
2017-11-29 09:40:07,006:INFO:root:Ready to return response
[2017-11-29 09:40:07 +0100] [79851] [ERROR] Error handling request /superset/csv/Hkqwzx3gM
Traceback (most recent call last):
  File "/opt/apps/superset/venv_superset_0.20.5/lib64/python3.4/site-packages/gunicorn/workers/sync.py", line 135, in handle
    self.handle_request(listener, req, client, addr)
  File "/opt/apps/superset/venv_superset_0.20.5/lib64/python3.4/site-packages/gunicorn/workers/sync.py", line 182, in handle_request
    resp.write(item)
  File "/opt/apps/superset/venv_superset_0.20.5/lib64/python3.4/site-packages/gunicorn/http/wsgi.py", line 341, in write
    self.send_headers()
  File "/opt/apps/superset/venv_superset_0.20.5/lib64/python3.4/site-packages/gunicorn/http/wsgi.py", line 337, in send_headers
    util.write(self.sock, util.to_bytestring(header_str, "ascii"))
  File "/opt/apps/superset/venv_superset_0.20.5/lib64/python3.4/site-packages/gunicorn/util.py", line 509, in to_bytestring
    return value.encode(encoding)
UnicodeEncodeError: 'ascii' codec can't encode character '\xfc' in position 206: ordinal not in range(128)
```

### Steps to reproduce
Rename a tab to, e.g., 'ü', run a query and try to expoprt it.

------------

I suspect that the filename generated from the tab label is the culprit.
If gunicorn can't handle noin-ascii characters, they would have to be replaced, I guess...

--- a/setup.py
+++ b/setup.py
@@ -75,6 +75,7 @@ setup(
         'sqlparse==0.2.3',
         'thrift>=0.9.3',
         'thrift-sasl>=0.2.1',
+        'unidecode>=0.04.21',
     ],
     extras_require={
         'cors': ['Flask-Cors>=2.0.0'],

--- a/superset/views/core.py
+++ b/superset/views/core.py
@@ -13,6 +13,7 @@ import re
 import time
 import traceback
 from urllib import parse
+from unidecode import unidecode
 
 from flask import (
     flash, g, Markup, redirect, render_template, request, Response, url_for,
@@ -2288,7 +2289,7 @@ class Superset(BaseSupersetView):
             csv = df.to_csv(index=False, **config.get('CSV_EXPORT'))
         response = Response(csv, mimetype='text/csv')
         response.headers['Content-Disposition'] = (
-            'attachment; filename={}.csv'.format(query.name))
+            'attachment; filename={}.csv'.format(unidecode(query.name)))
         logging.info('Ready to return response')
         return response
 

--- a/superset/views/core.py
+++ b/superset/views/core.py
@@ -13,7 +13,6 @@ import re
 import time
 import traceback
 from urllib import parse
-from unidecode import unidecode
 
 from flask import (
     flash, g, Markup, redirect, render_template, request, Response, url_for,
@@ -29,6 +28,7 @@ import pandas as pd
 import sqlalchemy as sqla
 from sqlalchemy import create_engine
 from sqlalchemy.engine.url import make_url
+from unidecode import unidecode
 from werkzeug.routing import BaseConverter
 from werkzeug.utils import secure_filename
 

